plugins/bitrot/transactions.go

```diff
@@ -97,3 +98,41 @@ error:
 	//TODO: Handle failure of scrubd. bitd should be stopped. Should it be handled in txn undo func
 	return err
 }
+
+func txnBitrotScrubOndemand(c transaction.TxnCtx) error {
+
+	var volname string
+	if err := c.Get("volname", &volname); err != nil {
+		c.Logger().WithError(err).WithField(
+			"key", "volname").Error("failed to get value for key from context")
+		return err
+	}
+
+	scrubDaemon, err := newScrubd()
+	if err != nil {
+		return err
+	}
+
+	c.Logger().WithFields(log.Fields{"volume": volname}).Info("Starting scrubber")
+
+	client, err := daemon.GetRPCClient(scrubDaemon)
+	if err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", volname).Error("failed to connect to scrubd")
+		return err
+	}
+
+	req := &brick.GfBrickOpReq{
+		Name: volname,
+		Op:   brick.OpNodeBitrot,
+	}
+	var rsp brick.GfBrickOpRsp
+	err = client.Call("BrickOp", req, &rsp)
```

Comment:
PR #537 fixes the RPC issue and is in master now. Please rebase. This call will now look like:

```go
client.Call("Brick.OpNodeBitrot", req, &rsp)
```

Reply:
dict unserialize is failing at scrub daemon end. Let me retest.

[2018-01-25 10:19:23.190595] W [dict.c:2835:dict_unserialize] (-->/usr/local/lib/libglusterfs.so.0(+0x72ab3) [0x7f28cd911ab3] -->/usr/local/sbin/glusterfs(glusterfs_handle_bitrot+0x1d4) [0x40d1a1] -->/usr/local/lib/libglusterfs.so.0(dict_unserialize+0xa7) [0x7f28cd8bed65] ) 0-dict: buf is null! [Invalid argument]

Reply:
Fixed! It was using dict and had serialize. But DictSerialize was in sunrpc package which was causing cycling dependency. Hence moved to package dict